homeassistant/components/epson/config_flow.py

```diff
@@ -0,0 +1,115 @@
+    }
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    epson_proj = epson.Projector(
+        data[CONF_HOST],
+        websession=async_get_clientsession(hass, verify_ssl=data.get(CONF_SSL, False)),
+        port=data[CONF_PORT],
+    )
+    _power = await epson_proj.get_property(POWER)
+    if not _power or _power == STATE_UNAVAILABLE:
+        raise CannotConnect
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for epson."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Epson options flow."""
+        return EpsonOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_config):
+        """Import a config entry from configuration.yaml."""
+        return await self.async_step_user(import_config)
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            if self.host_already_configured(user_input[CONF_HOST]):
+                return self.async_abort(reason="already_configured")
+            try:
+                await validate_input(self.hass, user_input)
+                return self.async_create_entry(
+                    title=user_input.pop(CONF_NAME), data=user_input
+                )
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    def host_already_configured(self, host):
+        """See if we already have a entry matching user input configured."""
+        existing_hosts = {
+            entry.data[CONF_HOST] for entry in self._async_current_entries()
+        }
+        return host in existing_hosts
+
+
+class CannotConnect(exceptions.HomeAssistantError):
+    """Error to indicate we cannot connect."""
+
+
+class EpsonOptionsFlowHandler(config_entries.OptionsFlow):
+    """Config flow options for Epson."""
+
+    def __init__(self, config_entry):
+        """Initialize Epson options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        return await self.async_step_user()
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        TIMEOUT_SCALE,
```

Comment:
Why do we allow this to be configured?

Reply:
epson documentation defines 3 timeouts:
```
PWR ON -> 40 secs,
PWR OFF -> 10,
CHANGE SOURCE -> 5,
OTHER -> 3
```

unfortunately, some projectors are slower and according to people who have them only way to get projector working is to increase this timeout. So I decided to just multiply timeouts by timeout scale.

https://github.com/home-assistant/core/issues/38743#issuecomment-678538835

Reply:
I don't like this option because we put the burden on the user. Could we detect what model we're communicating with and include a timeout scale for each model ?

Reply:
I tried this approach, but I could only get projector name (same as hostname so really not useful) using serial over tcp over port 3620. 
App doesn't have any info about model and even if I somehow I could detect model, I don't know which model is slower and why (one person reported that pretty new 4K projector is timing out for him).

Only way I can see is to drop this option for now and return to it if any author of the issues above will report back that the problem still persist with new version of lib and component.

Reply:
Let's try with the new version first. I am assuming the Android app also doesn't have a timeout option :)

Reply:
Android app for remote control the projector is just webview as projector is sharing website for remote control and during powering on, off it is not responding.

Reply:
This issue still persists for me with Epson 5040UB projector. Frequently turning "unavailable"